salt/states/pkg.py

```diff
@@ -1707,8 +1749,13 @@ def installed(
         if __grains__['os'] == 'FreeBSD':
             kwargs['with_origin'] = True
         new_pkgs = __salt__['pkg.list_pkgs'](versions_as_list=True, **kwargs)
+        if kwargs.get('resolve_capabilities', False) and 'pkg.list_provides' in __salt__:
+            new_caps = __salt__['pkg.list_provides'](**kwargs)
+        else:
+            new_caps = {}
         ok, failed = _verify_install(desired, new_pkgs,
-                                     ignore_epoch=ignore_epoch)
+                                     ignore_epoch=ignore_epoch,
+                                     new_caps=new_caps)
```

Comment:
I am still not sure why you often checking if `pkg.list_provides` method you've coded above shouldn't be introspected and loaded into a `__salt__`. But in case this is not the case, then just:
```python
ok, failed = _verify_install(desired, new_pkgs, ignore_epoch=ignore_epoch,
                             new_caps=list_provides(**kwargs))
```
The `def list_provides` should check inside the `resolve_capabilities` is `True`, instead of doing this across the entire code every time. This also will eliminate multiple `if/else` verifications.